gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFSInputStream.java

```diff
@@ -293,6 +287,8 @@ public synchronized void close() throws IOException {
             closeAPITrace(CLOSE_METHOD, stopwatch);
           }
 
+          streamStats.close();
+          seekStreamStats.close();
```

Comment:
what about closing the statistics stream in finally block?

Reply:
close() do not throw exception in normal case. Agree, it would be safer. Changed.